components/core/src/clp_s/ColumnWriter.hpp

```diff
@@ -141,7 +144,7 @@ class ClpStringColumnWriter : public BaseColumnWriter {
      * @param encoded_id
      * @return the encoded log dict id
      */
-    static clp::logtype_dictionary_id_t get_encoded_log_dict_id(uint64_t encoded_id) {
+    static clp::logtype_dictionary_id_t get_encoded_log_dict_id(encoded_log_dict_id_t encoded_id) {
         return static_cast<clp::logtype_dictionary_id_t>(encoded_id & cLogDictIdMask);
```

Comment:
it might sounds silly, but shall we do another cast here from uint64_t to clp::logtype_dictionary_id_t?